src/Microsoft.AspNetCore.Razor.Language/Intermediate/IntermediateNodeBuilder.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.AspNetCore.Razor.Language.Intermediate
 {
-    public abstract class IntermediateNodeBuilder
+    internal abstract class IntermediateNodeBuilder
```

Comment:
I don't want to make a bunch of change for change sake for something that's already working, tested and non-public.

@ajaybhargavb if you want to simplify this code go ahead, but it's not a priority